File: dashlane-project/Dashlane/src/main/java/com/dashlane/ui/menu/DashlaneMenuView.java
package com.dashlane.ui.menu;

import android.content.Context;
import android.util.AttributeSet;

import com.dashlane.ui.adapter.DashlaneRecyclerAdapter;
import com.dashlane.ui.menu.teamspace.TeamspaceAdapterItem;
import com.skocken.presentation.definition.Base;

import java.util.List;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;



public class DashlaneMenuView extends ScrimInsetsRecyclerView implements MenuDef.IView {

    private MenuDef.IPresenter mPresenter;
    private DashlaneRecyclerAdapter<MenuDef.Item> mMenuAdapter;

    public DashlaneMenuView(Context context) {
        this(context, null);
    }

    public DashlaneMenuView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public DashlaneMenuView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setLayoutManager(new LinearLayoutManager(context, RecyclerView.VERTICAL, false));
        MenuDef.IPresenter presenter = MenuComponent.Companion.invoke(context).getMenuPresenter();
        MenuDataProvider provider = new MenuDataProvider();
        if (presenter instanceof MenuPresenter) {
            ((MenuPresenter) (presenter)).setProvider(provider);
        }
        presenter.setView(this);
        setPresenter(presenter);
        setItemAnimator(null); 

        mMenuAdapter = new DashlaneRecyclerAdapter<>();
        mMenuAdapter.setOnItemClickListener((adapter, view, item, position) -> {
            if (item instanceof TeamspaceAdapterItem) {
                mPresenter.onTeamspaceSelected((TeamspaceAdapterItem) item);
            }
            item.doNavigation(provider.getMenuUsageLogger());
        });
        setAdapter(mMenuAdapter);
    }

    @Override
    public final void setPresenter(Base.IPresenter presenter) {
        mPresenter = (MenuDef.IPresenter) presenter;
    }

    @Override
    public void setItems(List<? extends MenuDef.Item> items) {
        mMenuAdapter.populateItems((List<MenuDef.Item>) items);
    }

    public void refresh() {
        mPresenter.refreshMenuList();
    }

}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/ui/menu/DashlaneMenuView.kt
package com.dashlane.ui.menu

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.LinearLayoutManager
import com.dashlane.ui.adapter.DashlaneRecyclerAdapter
import com.dashlane.ui.menu.MenuComponent.Companion.invoke
import com.dashlane.ui.menu.teamspace.TeamspaceAdapterItem
import com.skocken.presentation.definition.Base
import dagger.hilt.EntryPoint
import dagger.hilt.InstallIn
import dagger.hilt.android.EntryPointAccessors
import dagger.hilt.components.SingletonComponent

class DashlaneMenuView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : ScrimInsetsRecyclerView(context, attrs, defStyleAttr), MenuDef.IView {
    var presenter: MenuDef.IPresenter = invoke(context).menuPresenter
    private val menuAdapter: DashlaneRecyclerAdapter<MenuDef.Item>

    val dataProvider: MenuDataProvider =
        EntryPointAccessors.fromApplication(context, MenuEntryPoint::class.java).menuDataProvider

    init {
        layoutManager = LinearLayoutManager(context, VERTICAL, false)
        (presenter as? MenuPresenter)?.setProvider(dataProvider)
        presenter.setView(this)
        setPresenter(presenter)
        itemAnimator = null 
        menuAdapter = DashlaneRecyclerAdapter()
        menuAdapter.setOnItemClickListener { _, _, item: MenuDef.Item?, _ ->
            item ?: return@setOnItemClickListener
            if (item is TeamspaceAdapterItem) {
                presenter.onTeamspaceSelected(item)
            }
            item.doNavigation(dataProvider.menuUsageLogger)
        }
        adapter = menuAdapter
    }

    override fun setItems(items: List<MenuDef.Item>) {
        menuAdapter.populateItems(items)
    }

    override fun setPresenter(presenter: Base.IPresenter?) {
    }

    fun refresh() {
        presenter.refreshMenuList()
    }

    @InstallIn(SingletonComponent::class)
    @EntryPoint
    interface MenuEntryPoint {
        val menuDataProvider: MenuDataProvider
    }
}